### ISSUE TITLE
Suggestion: option to show/hide function parentheses in autocomplete entries

### ISSUE THREAD

**Issue by rodrigocfd** [rust-analyzer/rust-analyzer#9577]
In some past version (I don't remember which one), rust-analyzer started adding parentheses to functions in autocomplete suggestions. In the example below, parentheses are appended to function names:

* functions without arguments receive `()`;
* those with arguments receive `(...)`:

![Screnshot](https://i.imgur.com/c3MVHLX.png)

My suggestion is making these parentheses optional, so their rendering can be enabled or disabled by the user.

**Comment by mahdi-frms** [comment]
These are already optional (at least on VSCode):
simply add ```"rust-analyzer.completion.addCallParenthesis": false``` to your ```settings.json``` file and the parentheses are gone.

**Comment by flodiebold** [comment]
It'd be interesting if you could elaborate why you want to disable them, though.

**Comment by rodrigocfd** [comment]
@mahdi-frms Yes, `rust-analyzer.completion.addCallParenthesis` gets rid of the parentheses in the list. However, it *also* gets rid of the auto-added parentheses when you hit Enter. I believe these should be controlled by two different settings.

@flodiebold I have three reasons to justify why I don't need them:
1. Latest version brought #9494, which I personally consider *fantastic*, but now we have *a lot* of parentheses on each line;
2. We already have the second box with parentheses and parameters – in my screenshot, it's the box at the right of the autocompletion list, showing `fn(&self, i32)`.
3. We already have different icons with different colors to differentiate methods from fields.

**Comment by mahdi-frms** [comment]
Yes, I personally believe we should have separate settings for them.

**Comment by jeffs** [comment]
I'd like these parens to be optional at the time of autocompletion, rather than as a blanket setting.  In particular, they're a pain when I'm trying to pass a function as a callback, rather than trying to call the function directly.
